developer-workflow/extension-modules.rst

```diff
@@ -5,13 +5,679 @@
+   :emphasize-lines: 3
+
+   *@MODULE_BUILDTYPE@*
+   ...
+   @MODULE__FOO_TRUE@_foo _foo/_foomodule.c _foo/helper.c
+   ...
+   *shared*
+
+The ``@MODULE_<NAME_UPPER>_TRUE@<NAME>`` marker expects ``<NAME_UPPER>`` to
+be the upper-cased form of ``<NAME>``, where ``<NAME>`` has the same meaning
+as before (in our case, ``<NAME_UPPER>`` and ``<NAME>`` are ``_FOO`` and
+``_foo`` respectively). The marker is followed by the list of source files.
+
+If the extension module must be built as a *shared* module, put the
+``@MODULE__FOO_TRUE@_foo`` line after the ``*shared*`` marker:
+
+.. code-block:: text
+   :caption: :cpy-file:`Modules/Setup.stdlib.in`
+   :emphasize-lines: 4
+
+   ...
+   *shared*
+   ...
+   @MODULE__FOO_TRUE@_foo _foo/_foomodule.c _foo/helper.c
+
+Updating :cpy-file:`configure.ac`
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. add section about configuration variable afterwards
+
+* Locate the ``SRCDIRS`` variable and add the following line:
+
+  .. code-block:: text
+     :caption: :cpy-file:`configure.ac`
+     :emphasize-lines: 4
+
+     AC_SUBST([SRCDIRS])
+     SRCDIRS="\
+     ...
+     Modules/_foo \
+     ..."
+
+  .. note::
+
+     This step is only needed when adding new source directories to
+     the CPython project.
+
+* Find the section containing ``PY_STDLIB_MOD`` and ``PY_STDLIB_MOD_SIMPLE``
+  usages and add the following line:
+
+  .. code-block:: text
+     :caption: :cpy-file:`configure.ac`
+     :emphasize-lines: 3
+
+     dnl always enabled extension modules
+     ...
+     PY_STDLIB_MOD_SIMPLE([_foo], [-I\$(srcdir)/Modules/_foo], [])
+     ...
+
+  The ``PY_STDLIB_MOD_SIMPLE`` macro takes as arguments:
+
+  * the module name ``<NAME>`` used in :keyword:`import` statements,
+  * the compiler flags (CFLAGS), and
+  * the linker flags (LDFLAGS).
+
+  If the extension module may not be enabled or supported depending on the
+  host configuration, use the ``PY_STDLIB_MOD`` macro instead, which takes
+  as arguments:
+
+  * the module name ``<NAME>`` used in :keyword:`import` statements,
+  * a boolean indicating whether the extension is **enabled** or not,
+  * a boolean indicating whether the extension is **supported** or not,
+  * the compiler flags (CFLAGS), and
+  * the linker flags (LDFLAGS).
+
+  For instance, enabling the :mod:`!_foo` extension on Linux platforms, but
+  only providing support for 32-bit architecture, is achieved as follows:
+
+  .. code-block:: text
+     :caption: :cpy-file:`configure.ac`
+     :emphasize-lines: 2, 3
+
+     PY_STDLIB_MOD([_foo],
+                   [test "$ac_sys_system" = "Linux"],
+                   [test "$ARCH_RUN_32BIT" = "true"],
+                   [-I\$(srcdir)/Modules/_foo], [])
+
+  More generally, the status of the extension is determined as follows:
+
+  +-----------+-----------------+----------+
+  | Enabled   | Supported       | Status   |
+  +===========+=================+==========+
+  | true      | true            | yes      |
```

Comment:
Perhaps available instead of yes.

```suggestion
  | true      | true            | available      |
```

Reply:
Actually, for this one, I will to disagree as `./configure` says:

```
dnl Check for stdlib extension modules
dnl PY_STDLIB_MOD([NAME], [ENABLED-TEST], [SUPPORTED-TEST], [CFLAGS], [LDFLAGS])
dnl sets MODULE_$NAME_STATE based on PY_STDLIB_MOD_SET_NA(), ENABLED-TEST,
dnl and SUPPORTED_TEST. ENABLED-TEST and SUPPORTED-TEST default to true if
dnl empty.
dnl    n/a: marked unavailable on platform by PY_STDLIB_MOD_SET_NA()
dnl    yes: enabled and supported
dnl    missing: enabled and not supported
dnl    disabled: not enabled
``` 

and 

``` 
  AS_IF([test "$modstate" != "n/a"], [
    AS_IF([m4_ifblank([$2], [true], [$2])],
       [AS_IF([m4_ifblank([$3], [true], [$3])], [modstate=yes], [modstate=missing])],
       [modstate=disabled])
  ])
```

Reply:
Sounds good. Thanks for clarifying.